Value from vec

SergioBenitez opened the pull request:
The first commit is an optimization, since what I was assume was originally an optimization attempt is more expensive than keeping simply keeping the value around, but feel free to drop it if you'd like. The second is rather important to avoid re-allocating vectors that already contain `Value`s when creating a `Value`.

mitsuhiko reviewed (changes requested):
I think I would be favoring removing the `From<T: Into<Value>>` impl instead as you proposed.

SergioBenitez commented:
Alright, cleaned up a few more `From` impls. I actually kept the `From<T: into<Value>> Vec<T>` because otherwise, you can't return things like `Vec<String>` from a filter/function/etc. But, I added a `From<Arc<Vec<Value>>>` and a `FromIterator` which both allow cost-free creations of the `Seq`.

mitsuhiko commented:
One thing I now overlooked is that the Value increased in size from 24 bytes to 32 bytes as the i128/u128 is now inlined. I will revisit this separately.
